backend/modules/parsers/unstructured_io.py

```diff
@@ -0,0 +1,92 @@
+import requests
+from langchain.docstore.document import Document
+from requests.adapters import HTTPAdapter, Retry
+
+from backend.logger import logger
+from backend.modules.parsers.parser import BaseParser
+from backend.settings import settings
+
+
+class UnstructuredIoParser(BaseParser):
+    """
+    UnstructuredIoParser is a parser class for extracting text from unstructured input.
+    """
+
+    supported_file_extensions = [
+        ".txt",
+        ".eml",
+        ".msg",
+        ".xml",
+        ".html",
+        ".md",
+        ".rst",
+        ".json",
+        ".rtf",
+        ".jpeg",
+        ".png",
+        ".doc",
+        ".docx",
+        ".ppt",
+        ".pptx",
+        ".pdf",
+        ".odt",
+        ".epub",
+        ".csv",
+        ".tsv",
+        ".xlsx",
+    ]
+
+    def __init__(self, max_chunk_size: int = 2000, *args, **kwargs):
+        """
+        Initializes the UnstructuredIoParser object.
+        """
+        self.max_chunk_size = max_chunk_size
+        self.session = requests.Session()
+        self.retry_strategy = Retry(
+            total=3,
+            backoff_factor=1,
+            status_forcelist=[429, 500, 502, 503, 504],
+            allowed_methods=["POST"],
+        )
+        self.adapter = HTTPAdapter(max_retries=self.retry_strategy)
+        self.session.mount("https://", self.adapter)
+        self.session.mount("http://", self.adapter)
+        self.headers = {
+            "accept": "application/json",
+        }
+        if settings.UNSTUCTURED_IO_API_KEY:
+            self.headers["unstructured-api-key"] = settings.UNSTUCTURED_IO_API_KEY
+
+    async def get_chunks(self, filepath: str, metadata: dict, *args, **kwargs):
+        """
+        Asynchronously extracts text from unstructured input and returns it in chunks.
+        """
+        final_texts = list()
+        try:
+            with open(filepath, "rb") as f:
+                # Define files payload
+                files = {"files": f}
+                data = {
+                    "strategy": "auto",
+                    "languages": ["eng", "hin"],
+                    "chunking_strategy": "by_title",
+                    "max_characters": self.max_chunk_size,
+                }
+
+                # Send POST request
+                response = self.session.post(
+                    settings.UNSTUCTURED_IO_URL.rstrip("/") + "/general/v0/general",
+                    headers=self.headers,
+                    files=files,
+                    data=data,
+                )
+
+            parsed_data = response.json()
+            for payload in parsed_data:
+                text = payload["text"]
+                metadata = payload["metadata"]
+                final_texts.append(Document(page_content=text, metadata=metadata))
+            return final_texts
+        except Exception as e:
+            logger.exception(f"Final Exception: {e}")
+            return final_texts
```

Comment:
This is too aggressive error handling in my opinion, the caller should decide what they want to do this errors
This is general comment across all parsers that we maintain